### DISCUSSION TITLE
How to let  the app to minimize, and stay in the system tray?

### DISCUSSION THREAD

**Discussion by fayfive** [fyne-io/fyne#3237] [Q&A] [answered]
How to let  the app to minimize, and stay in the system tray? It seem no this API.

**Comment by Jacalz** [comment]
You can set up a close intercept on the main window to make sure that the application is hidden instead of closed when pressing the close button.

Like this:
```go
a := app.New()
w := a.NewWindow("Window Name")

if _, ok := a.(desktop.App); ok {
    w.SetCloseIntercept(w.Hide) // don't close the window if system tray is used
}
```

You of course also need to create the system tray within that if-statement but I assume that you have that code already.

**Reply by fayfive** [reply]
I know this method. But I want to click the window "-" to stay on the tray not only the "X"

**Reply by Jacalz** [reply]
Doesn’t a minimised application already show the tray? Or are you perhaps asking for it to disappear from the list of opened windows and stay in the systemtray instead?

**Reply by Jacalz** [reply]
If you are talking about the latter, I think https://github.com/fyne-io/fyne/issues/3156 like is what you want.

**Reply by andydotxyz** [reply]
Most OS will not hide a window that is minimised. We cannot override that.

**Comment by Jacalz** [comment, marked as answer]
This might be a duplicate of #3156 come to think of it.

**Reply by fayfive** [reply]
Yes! If I click the "-", the app will do it like that.
I can listen the minimize event on MFC

**Reply by Jacalz** [reply]
Cool. I think that’s as good as we can get it until the issue is fixed then. Feel free to mark as solved if you want to.
